### DISCUSSION TITLE
Issues after upgrading to clamav 0.104.0

### DISCUSSION THREAD

**Discussion by ienev** [Homebrew/discussions#2113] [Casks] [answered]
`brew upgrade` made clamav upgrade from 0.103.3 to 0.104.0

Now `freshclam` and `clamscan` exit with an error:

```
➜  ~ freshclam -v

ERROR: Can't open/parse the config file /usr/local/Cellar/clamav/0.104.0/etc/freshclam.conf
```

I'm not sure why it searches it in this location. The previous versions used the file under `/usr/local/etc/clamav`

```
➜  ~ ls -al /usr/local/etc/clamav/freshclam.conf

-rw-r--r--  1 user  user  6443 Jul 16  2019 /usr/local/etc/clamav/freshclam.conf
```

And here is how `clamscan` fails:

```
➜  ~ clamscan

LibClamAV Error: cl_load(): No such file or directory: /usr/local/Cellar/clamav/0.104.0/share/clamav
ERROR: Can't get file status

----------- SCAN SUMMARY -----------
Known viruses: 0
Engine version: 0.104.0
Scanned directories: 0
Scanned files: 0
Infected files: 0
Data scanned: 0.00 MB
Data read: 0.00 MB (ratio 0.00:1)
Time: 0.347 sec (0 m 0 s)
Start Date: 2021:09:08 21:47:10
End Date:   2021:09:08 21:47:11
```

```
➜  ~ ls -al /usr/local/Cellar/clamav/0.104.0/share

total 0
drwxr-xr-x   4 user  user  128 Aug 28 01:41 .
drwxr-xr-x  14 user  user  448 Sep  8 21:46 ..
drwxr-xr-x   3 user  user   96 Aug 28 01:41 doc
drwxr-xr-x   5 user  user  160 Aug 28 01:41 man
```

**Comment by cho-m** [comment, marked as answer]
ClamAV's build system changed between those releases (from Autotools to CMake) and the formula installation needed a rewrite to support this: https://github.com/Homebrew/homebrew-core/pull/84636

Previously, in Autotools build, there was a `--sysconfdir=#{etc}/clamav`, which referenced Homebrew's `etc` directory. This may need to be specified in CMake build via `-DAPP_CONFIG_DIRECTORY=#{etc}/clamav` as default seems to be a relative path within `CMAKE_INSTALL_PREFIX` (which is formula's install directory).
https://github.com/Cisco-Talos/clamav/blob/main/INSTALL.md#customizing-the-install-directories

---

Not sure how previous install looked for `DATABASE_DIRECTORY`, which is currently set to `<path-to-formula>/share/clamav`. It may be an empty directory that was removed when creating bottle and for some reason it cannot be created by `clamscan`.

It may be that setting up database failed in other command, so `clamav` also failed.

**Reply by cho-m** [reply]
Opened a PR https://github.com/Homebrew/homebrew-core/pull/84885 to fix the 1st issue. I think fixing this should allow creating database to fix 2nd issue, but I haven't tried.

I don't have config file, but at least running `freshclam` command outputs previous file path:
```console
❯ freshclam -v
ERROR: Can't open/parse the config file /usr/local/etc/clamav/freshclam.conf
```

**Reply by ienev** [reply]
Yep, that worked locally, thanks.

**Comment by Oclair** [comment]
If this is unclear for you how it was for me, then how to fix the issue you need to do the following:
mkdir /usr/local/Cellar/clamav/0.104.1/share/clamav

edit 
/usr/local/etc/clamav/freshclam.conf
and
/usr/local/etc/clamav/clamd.conf

ensure the line with "Example" looks like
#Example
